minari/data_collector/data_collector.py

```diff
@@ -250,8 +266,8 @@ def reset(
         assert STEP_DATA_KEYS.issubset(step_data.keys())
 
         # If last episode in global buffer has saved steps, we need to check if it was truncated or terminated
-        # If not, then we need to auto-truncate the episode
-        if len(self._buffer[-1]["actions"]) > 0:
+        # If the last element in the buffer is not an empty dictionary, then we need to auto-truncate the episode.
+        if self._buffer[-1]:
```

Comment:
If the episode ends via the wrapped environment returning `terminated=True` when calling step, an empty dictionary is added to `self._buffer` so `self.buffer[-1]` returns `False`. As far as I can tell, the behavior is correct.

I agree that it would be good to add an explicit test for this behavior, I can add a basic one.

Reply:
I added a line to test for it in an existing dataset creation test https://github.com/Farama-Foundation/Minari/pull/77/commits/58587495c0be99e97d44fe915ba0161876af1425

Reply:
> If the episode ends via the wrapped environment returning terminated=True when calling step, an empty dictionary is added to self._buffer so self.buffer[-1] returns False. As far as I can tell, the behavior is correct.

I see; I find confusing that sometimes we add an "initialized" dictionary and other an empty one.
This probably adds to the other comments about a refactor that we can do after this is merged.